user/caribou/module/src/DSO9254AEventConverter.cc

```diff
@@ -0,0 +1,377 @@
+#include "CaribouEvent2StdEventConverter.hh"
+#include "utils/log.hpp"
+#include "CLICTDFrameDecoder.hpp"
+#include "TFile.h"
+#include "TH1D.h"
+#include "time.h"
+
+using namespace eudaq;
+
+namespace{
+  auto dummy0 = eudaq::Factory<eudaq::StdEventConverter>::
+  Register<DSO9254AEvent2StdEventConverter>(DSO9254AEvent2StdEventConverter::m_id_factory);
+}
+
+uint64_t timeConverter( std::string date, std::string time );
+
+bool DSO9254AEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::StandardEventSP d2, eudaq::ConfigurationSPC conf) const{
+  
+
+  auto ev = std::dynamic_pointer_cast<const eudaq::RawEvent>(d1);
+  // No event
+  if(!ev) {
+    return false;
+  }
+
+
+  // load parameters from config file
+  double pedStartTime = conf->Get("pedStartTime", 0 ); // integration windows in [ns]
+  double pedEndTime   = conf->Get("pedEndTime"  , 0 );
+  double ampStartTime = conf->Get("ampStartTime", 0 );
+  double ampEndTime   = conf->Get("ampEndTime"  , 0 );
+  bool generateRoot   = conf->Get("generateRoot", 0 ); 
+  std::cout << "Loaded parameters from configuration file." << std::endl
+	    << "  pedStartTime = " << pedStartTime << std::endl
+	    << "  pedEndTime   = " << pedEndTime   << std::endl
+	    << "  ampStartTime = " << ampStartTime << std::endl
+	    << "  ampEndTime   = " << ampEndTime   << std::endl
+	    << "  generateRoot = " << generateRoot << std::endl << std::endl;
+  
+
+  // Data container:
+  caribou::pearyRawData rawdata;
+
+  // internal containers
+  std::vector<std::vector<std::vector<double>>> wavess;
+  std::vector<std::vector<double>> peds;
+  std::vector<std::vector<double>> amps;
+  std::vector<double> time;
+  uint64_t timestamp;
+
+  // waveform parameters
+  std::vector<uint64_t> np;
+  std::vector<double> dx;
+  std::vector<double> x0;
+  std::vector<double> dy;
+  std::vector<double> y0;
+
+  
+  // generate rootfile to write waveforms as TH1D
+  TFile * histoFile = nullptr;
+  if( generateRoot ){
+    histoFile = new TFile( "waveforms.root", "UPDATE" ); // UPDATE?
+    if(!histoFile) {
+      LOG(ERROR) << "ERROR: " << histoFile->GetName() << " can not be opened";
+      return false;
+    }
+  }
+						       
+  
+  // Retrieve data from event
+  if(ev->NumBlocks() == 1) {
```

Comment:
you check for number of blocks == 1 but don't do anything if they're *not* 1. Is this intended? Then data is simply skipped without notification or anything?